**Testes de spliting com os dados de treino com Random Forest**



In [1]:
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score, matthews_corrcoef

from google.colab import drive

drive.mount('/gdrive')
%cd /gdrive/MyDrive/rdc/tei

Mounted at /gdrive
/gdrive/MyDrive/rdc/tei


In [2]:
import pandas as pd

train_df = pd.read_csv('train.csv')
test_df = pd.read_csv('test.csv')

In [3]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()

train_df = train_df.drop('id', axis=1)

for col in train_df.columns:
  if col != 'class':
    le.fit(train_df[col])
    train_df[col] = le.transform(train_df[col])

for col in test_df.columns:
    if col != 'id':
      le.fit(test_df[col])
      test_df[col] = le.transform(test_df[col])

Spliting de 80% de treino e 20% de validação

In [4]:
X = train_df.drop('class', axis=1)
y = train_df['class']
X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.2, random_state=42)

In [5]:
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)

RandomForestClassifier(random_state=42)

In [6]:
y_pred = model.predict(X_val)

In [7]:
accuracy = accuracy_score(y_val, y_pred)
print(f'Accuracy: {accuracy}')

mcc = matthews_corrcoef(y_val, y_pred)
print(f'MCC: {mcc}')

Accuracy: 0.9920130127416429
MCC: 0.9838883796886083


Accuracy: 0.9920130127416429  
MCC: 0.9838883796886083  
Score Kaggle: 0.30641

Spliting de 50% para treino e para validação

In [17]:
X = train_df.drop('class', axis=1)
y = train_df['class']
X_train, X_val, y_train, y_val = train_test_split(X, y, test_size=0.5, random_state=42)

In [14]:
X_train.shape

(1558472, 20)

In [15]:
model = RandomForestClassifier(random_state=42)
model.fit(X_train, y_train)

RandomForestClassifier(random_state=42)

In [19]:
y_pred = model.predict(X_val)

In [20]:
accuracy = accuracy_score(y_val, y_pred)
print(f'Accuracy: {accuracy}')

mcc = matthews_corrcoef(y_val, y_pred)
print(f'MCC: {mcc}')

Accuracy: 0.9919940865193044
MCC: 0.9838500293035001


Accuracy: 0.9919940865193044  
MCC: 0.9838500293035001  
Score Kaggle: 0.28282

In [9]:
test_predictions = model.predict(test_df.drop('id', axis=1))

test_df['class'] = test_predictions
test_df[["id","class"]].to_csv("novoo.csv", index=False)

In [22]:
# para rodar pela segunda vez e reaproveitar o df_test
test_predictions = model.predict(test_df.drop('id', axis=1).drop('class', axis=1))

test_df['class'] = test_predictions
test_df[["id","class"]].to_csv("fooo.csv", index=False)